Sprint-1/fix/median.js

```diff
@@ -3,9 +3,26 @@
 // If you're in the Sprint-1 directory, you can run `npm test -- fix` to run the tests in the fix directory
 
 function calculateMedian(list) {
-  const middleIndex = Math.floor(list.length / 2);
-  const median = list.splice(middleIndex, 1)[0];
-  return median;
+
+  // 〰️ Create a copy of the input list to avoid modifying the original array
+  const copiedList = [...list];
+
+  // 〰️ Calculate the index of the middle element
+  const middleIndex = Math.floor(copiedList.length / 2)
+
+  // 〰️ If the list length is odd, return the middle element
+  if (copiedList.length % 2 === 1) {
+    const median = copiedList.splice(middleIndex, 1)[0];
```

Comment:
There is a simpler and more direct syntax to access an array element at a particular position.

Reply:
Yes, that's right. There is a simpler way to access an array element, and I’ve improved my code accordingly. Previously, I used splice to extract the middle element, which introduced unnecessary complexity. Instead, I removed the splice operation and replaced it with direct access using copiedList[middleIndex]. This is simpler and avoids modifying the array unnecessarily.
Thank you for pointing this out. It helped me make my code cleaner and more efficient! 😊